### PR TITLE
RequireJS migration: custom data fields

### PR DESCRIPTION
**Pull request by orangejenny**
This makes feature flags available in requirejs by moving them to initial page data.

Review by commit.

@dannyroberts / @millerdev

### CODE REVIEW

File: corehq/util/context_processors.py [hunk 1]
```diff
@@ -115,6 +115,19 @@ def js_api_keys(request):
...
+def js_toggles(request):
+    if not hasattr(request, 'couch_user') or not request.couch_user:
+        return {}
+    if not hasattr(request, 'project'):
+        return {}
+    from corehq import toggles, feature_previews
+    domain = request.project.name
+    return {
+        'TOGGLES_DICT': toggles.toggle_values_by_name(username=request.couch_user.username, domain=domain),
+        'PREVIEWS_DICT': feature_previews.preview_values_by_name(domain=domain)
```
Comment (millerdev) [inline corehq/util/context_processors.py]: nit: what do you think of changing these keys to lower-case? That would make it slightly easier to grep for them to see where they are referenced.

**Review by millerdev — APPROVED** [review]
It looks like all caching has been remove from toggles and preview calculations. Given the title of commit 98bbfd531cc9a0e21aa857fe2de87bd496f1f4e2, I'm assuming that was intentional. Was caching creating a problem, or was it not needed, or something else? Is there a chance we will need to add it back?

File: corehq/util/context_processors.py (hunk 1 above)
Comment (orangejenny) [reply corehq/util/context_processors.py]: sure => 301f5986f64

**Comment by orangejenny** [comment]
Yes, the caching removal was intentional. It was for the CDN/browser; it added cache busting GET params to the script tag for toggles.js. Now that toggles.js doesn't exist, there's no need for it.

**Pull request merged**